src/server/string_family.cc

```diff
@@ -501,10 +500,46 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
   return builder->SendSetSkipped();
 }
 
+OpResult<void> StringFamily::SetGeneric(ConnectionContext* cntx, SetCmd::SetParams sparams,
+                                        std::string_view key, std::string_view value) {
+  DCHECK(cntx->transaction);
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    SetCmd sg(t->GetOpArgs(shard));
+    return sg.Set(sparams, key, value);
+  };
+  return cntx->transaction->ScheduleSingleHop(std::move(cb));
+}
+
 void StringFamily::SetEx(CmdArgList args, ConnectionContext* cntx) {
   SetExGeneric(true, std::move(args), cntx);
 }
 
+void StringFamily::SetNx(CmdArgList args, ConnectionContext* cntx) {
+  // This is the same as calling the "Set" function, only in this case we are
+  // changing the value only if the don't have this key already, if it does exists
```

Comment:
changing the value only if the don't have this key already -> 
change the value only if the key does not exist. Otherwise,

Reply:
OK